node/src/main.rs

```diff
@@ -866,70 +880,209 @@ async fn main() {
+        if network_arg.starts_with("wss://")
+            || network_arg.starts_with("http://")
+            || network_arg.starts_with("https://")
+        {
+            return Err(format_err!(
+                "Is your Ethereum node string missing a network name? \
                      Try 'mainnet:' + the Ethereum node URL."
+            ));
+        } else {
+            // Parse string (format is "NETWORK_NAME:NETWORK_CAPABILITIES:URL" OR
+            // "NETWORK_NAME::URL" which will default to NETWORK_CAPABILITIES="archive,traces")
+            let split_at = network_arg.find(':').ok_or_else(|| {
+                return format_err!(
+                    "A network name must be provided alongside the \
+                         Ethereum node location. Try e.g. 'mainnet:URL'."
+                );
+            })?;
+
+            let (name, rest_with_delim) = network_arg.split_at(split_at);
+            let rest = &rest_with_delim[1..];
+            if name.is_empty() {
+                return Err(format_err!(
+                    "Ethereum network name cannot be an empty string"
                 ));
-            } else {
-                // Parse string (format is "NETWORK_NAME:URL")
-                let split_at = network.find(':').ok_or_else(|| {
-                    return format_err!(
-                        "A network name must be provided alongside the \
+            }
+
+            let url_split_at = rest.find(":").ok_or_else(|| {
+                return format_err!(
+                    "A network name must be provided alongside the \
                          Ethereum node location. Try e.g. 'mainnet:URL'."
-                    );
-                })?;
+                );
+            })?;
 
-                let (name, loc_with_delim) = network.split_at(split_at);
-                let loc = &loc_with_delim[1..];
+            let (network_capabilities_str, url_str) = rest.split_at(url_split_at);
+            let (url, network_capabilities) =
+                if vec!["http", "https", "ws", "wss"].contains(&network_capabilities_str) {
+                    (
+                        rest,
+                        NodeCapabilities {
+                            archive: true,
+                            traces: true,
+                        },
+                    )
+                } else {
+                    let capabilities = network_capabilities_str.split(',').collect::<Vec<&str>>();
+                    let invalid_capabilities: Vec<&&str> = capabilities
+                        .iter()
+                        .filter(|capability| !vec!["archive", "traces"].contains(capability))
+                        .collect();
+                    if !invalid_capabilities.is_empty() {
+                        return Err(format_err!(
+                            "Invalid Ethereum node capability supplied: {:?}",
+                            invalid_capabilities
+                        ));
```

Comment:
This would for a nice `FromStr` implementation for `NodeCapabilities`. Not a show-stopper but that would be nice to clean up in a follow-up PR.

I also still see the variable name `network_capabilities` here. ;)

Reply:
Well, ok, here it is:
```rust
impl FromStr for NodeCapabilities {
    type Err = Error;

    fn from_str(s: &str) -> Result<Self, Self::Err> {
        let capabilities: Vec<&str> = str.split(":").collect();
        Ok(NodeCapabilities {
            archive: capabilities.iter().find(cap => cap.eq("archive")).is_some(),
            traces: capabilities.iter().find(cap => cap.eq("traces")).is_some(),
        })
    }
}